### PR TITLE
Use extensions to configure plugin instead of project properties

### PR DESCRIPTION
**Pull request by bipicoppernic**
Remove the need of using command line parameters. Used gradle extensions instead. 

Remove also the dependency to github. We can use gitlab or an other git host.

### CODE REVIEW

**Comment by drapp** [comment]
Thanks, I couldn't get extensions working back when I first worked on this, no idea why

File: version.gradle [hunk 1]
```diff
@@ -1 +1 @@
-version='1.0.1'
+version='1.1.0'
```
Comment (drapp) [inline version.gradle]: this is backwards incompatible, so 2.0.0 is in order

File: src/main/groovy/com/layer/gradle/gitrepo/GitRepoPlugin.groovy [hunk 2]
```diff
@@ -14,10 +14,13 @@ class GitRepoPlugin  implements Plugin<Project> {
...
     void apply(Project project) {
 
+		project.extensions.create("repoconfig",GitRepoPluginExtension)
+		project.extensions.create("taskconfig",PublishTaskExtension)
+
         // allow declaring special repositories
         if (!project.repositories.metaClass.respondsTo(project.repositories, 'github', String, String, String, String, Object)) {
             project.repositories.metaClass.github = { String org, String repo, String branch = "master", String type = "releases", def closure = null ->
-                String gitUrl = gitCloneUrl(org, repo)
+                String gitUrl = gitCloneUrl(project)
```
Comment (drapp) [inline src/main/groovy/com/layer/gradle/gitrepo/GitRepoPlugin.groovy]: this right here is configuring the `github` dependency thing, so it's deliberately dependent on github. Right below there's a plain old `git` version where you specify the clone url. I think that's what you want. You can have multiple different git repo dependencies, so it won't work if they read the same values from the same `repoconfig` extension. The extensions will be useful for the publish task

File: src/main/groovy/com/layer/gradle/gitrepo/GitRepoPlugin.groovy [hunk 3]
```diff
@@ -14,10 +14,13 @@ class GitRepoPlugin  implements Plugin<Project> {
 
     void apply(Project project) {
 
+		project.extensions.create("repoconfig",GitRepoPluginExtension)
```
Comment (drapp) [inline src/main/groovy/com/layer/gradle/gitrepo/GitRepoPlugin.groovy]: any reason you're not using camelcase?

**Comment by drapp** [comment]
I'm tweaking this locally, if you don't object, I may want to merge my modified version in a day or so

**Comment by bipicoppernic** [comment]
Hi,

You can modify it like you want, add camel case or not. It's yours !

File: src/main/groovy/com/layer/gradle/gitrepo/GitRepoPlugin.groovy (hunk 3 above)
Comment (bipicoppernic) [inline src/main/groovy/com/layer/gradle/gitrepo/GitRepoPlugin.groovy]: Only the habit to use little case in extensions.

**Pull request merged**